src/app/courses/courses.component.ts

```diff
@@ -23,20 +23,43 @@ export class CoursesComponent implements OnInit {
     public orderByName: OrderByPipe,
     public filterSearch: FilterSearchPipe
   ) {
-    this.courses = this.coursesService.getAll(this.page, this.displayLimit);
+    var _this = this;
+    this.coursesService.getCoursesFromServer().subscribe({
```

Comment:
same here - getAll is clear enough.
please, use `pipe` here as follows:

```typescript
this.coursesService.getCoursesFromServer().pipe(
take(1),
tap(data => this.coursesService.setList(data)), 
tap(data => {
  this.courses = this.coursesService.getAll(this.page, this.displayLimit);
}),
catchError(msg => console.log('Error is getting course fetching: ', msg))
).subscribe()
```

Reply:
done